mlpipeline/pipeline/pipeline.py

```diff
@@ -0,0 +1,135 @@
+import logging
+from ..utils import get_pipeline_message
+
+##
+L = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+##
+
+
+class Pipeline:
+    """
+    Executes components in topological order and obtains the result
+
+    ...
+
+    Methods
+    -------
+    execute(input: dict)
+        obtains result
+    """
+
+    def __init__(self, name: str, inputs: set, outputs: set, components: dict, running_order: list):
```

Comment:
Using a dataclass would be a way to remove this boilerplate constructor.

Reply:
thanks for pointing me on it, this concept is very useful